### PR TITLE
Align the training callbacks parameter

### LINKED ISSUE #166: Retrieve more data out of the training loop
Hey! I'm looking into hooking up training a YOLOX model to [mlflow](https://mlflow.org/) and so far I was only able to log the final metrics defined here https://github.com/LibreYOLO/libreyolo/blob/449f9781ed1e1c574f7bce6c4f4f8efdb0b54c54/libreyolo/training/trainer.py#L402-L411

`epoch_losses` already contains the loss curve. This is good, but I was curious if we could consider more data points like the LR curve, mAP50/mAP50-95 per epoch and so on. I don't know if this means returning these once the training session ends or exposing a hook we can tap into after each epoch. What do you think?

Happy to help implement anything!

### PR DESCRIPTION
**Pull request by Magellol**
Closes #166 

Apologies for the delay! I've tried aligning the `callbacks` being passed through from the various models all the way through the `BaseTrainer`. Let me know if I missed anything.

### CODE REVIEW

File: libreyolo/training/config.py [hunk 1]
```diff
@@ -135,6 +137,7 @@ class TrainConfig:
     log_interval: int = 10
     seed: int = 0
     allow_download_scripts: bool = False
+    callbacks: TrainCallbacks = None
```
[the remark this answers is not in the document]
Comment (Magellol) [reply libreyolo/training/config.py]: I'm actually wondering if putting the `callbacks` in `TrainConfig` is a good idea after all. Shouldn't it be serializable? (e.g YAML config, etc)

File: libreyolo/training/config.py [hunk 2]
```diff
@@ -1,5 +1,7 @@
 """Training configuration dataclasses for LibreYOLO."""
 
+from libreyolo.training import TrainCallbacks
+
 import logging
```
[the remark this answers is not in the document]
Comment (Magellol) [reply libreyolo/training/config.py]: This should be fixed in https://github.com/LibreYOLO/libreyolo/pull/464/commits/a1e9bd10f8c80c9cd2fe10a95a81266613409dc2

**Comment by EHxuban11** [comment]
Hello @magellol, I hope you are well! Perfect, today I added the hook system for certain models but forgot the classifiers. I see that you added type hints, that's perfect.  Thank you !
